pup-0006.md

```diff
@@ -0,0 +1,101 @@
+the existing committers decided to stop that practice and instead document an open process.
+Engineers hired on the Pulp team since Oct '17 have not received commit bit.
+
+We have not yet documented an open process of which to give a proven contributor commit access to
+all or a portion of the codebase.
+
+
+## Scope
+
+This formally applies to the following core repositories and core-team maintained plugins:
+
+Core Repos:
+* pulp/pulp
+* pulp/devel
+* pulp/pulp-ci
+
+Plugin Repos:
+* pulp/pulp_file
+* pulp/pulp_python
+* pulp/pulp_rpm
+* pulp/crane
+* pulp/pulp_docker
+* pulp/pulp_puppet
+* pulp/pulp_ostree
+
+This policy is intended as the default for all software in the Pulp ecosystem unless explicitly
+stated otherwise.
+
+
+## Criteria
+
+Overall the candidate must have demonstrated commitment and care towards the needs of all Pulp
+users and not only their own interests. Also they must have the experience to be trusted with major
+aspects of Pulp functionality in that area.
+
+These requirements are somewhat vague by design and leave the decision to the judgement of the
+existing developers. Users rely on the judgement of developers already, so to rely on that judgement
+to accept new contributors is reasonable. Anyone who specifically wants to get more involved should
+approach the core devs about mentorship.
+
+
+## Process
+
+A nomination or self-nomination can be sent to the pulp-dev email list and must contain state the
+following:
+
+* Name and email of the nominee. This can be a self-nomination.
+* Repositories being requested
+* [Specific paths](https://blog.github.com/2017-07-06-introducing-code-owners/) or the entire repo
+* Vote end date. Must be 7 calendar days from now.
+
+To pass, a unanimous +1 vote is required from other committers in that area. Specifically:
+
+* Adding a committer to a subsystem requires 100% vote participation from all existing maintainers
+  of that subsystem.
+* Creating a first owner of a subsystem requires 100% vote participation from that repository's
+  committers.
+* Adding a committer for a repository requires 100% vote participation from that repository's
+  committers. 
+
+If a nomination does not pass, the candidate may be re-nominated after a cool down period of 60
+days.
+
+
+## Drawbacks
+
+A single developer can block a new candidate. This could limit contribution unnecessarily and
+possibly create fairness issues.
+
+Candidates might get waived through because casting a blocking vote can be uncomfortable.
+
+
+## Alternatives
+
+One alternative is to never add new committers. That is not a good option.
+
+Another alternative is to use hard metrics such as time involved, feature planned and implemented,
+bugs fixed, etc. Overall hard metric system can be gamed which is a major downside. Also a metrics
+based process does not help unify the committer base like a unanimous vote would.
+
+
+## Unresolved Questions
+
+How will the commit bit be removed?
```

Comment:
I think this is an important issue and deserves addressing in this document.

Reply:
this specific pup is about 'commit bit assignment', i would expect to be written a separate pup which would address commit bit removal and relieve.

Reply:
I would be fine with either but even if it is a separate PUP, we should get it over and done with now I think.

Reply:
Assigning commit bit criteria change indeed needn't imply change in criteria removal.
Still I'd find it more safe against the: "oh I didn't realize the implications" kind of errors if both assigning and relieve criteria are specified in a single doc.
We're most likely talking about <= +20 slocs worth of change fwiw.